incubator/hnc/test/e2e/demo_test.go

```diff
@@ -93,8 +93,91 @@ var _ = Describe("Demo", func() {
+		RunShouldContain("Welcome to nginx!", 10, 
+			"kubectl run client -n", nsTeamA, "-i --image=alpine --restart=Never --rm -- sh -c", "\"wget -qO- --timeout 2 http://s2.service-2\"")
+		RunShouldContain("Welcome to nginx!", 10, 
+			"kubectl run client -n", nsTeamB, "-i --image=alpine --restart=Never --rm -- sh -c", "\"wget -qO- --timeout 2 http://s2.service-2\"")
+
+		// create a default network policy that blocks any ingress from other namespaces 
+		policy := "" +
+			"kind: NetworkPolicy" + "\n" +
+			"apiVersion: networking.k8s.io/v1" + "\n" +
+			"metadata:" + "\n" +
+			"  name: deny-from-other-namespaces" + "\n" +
+			"  namespace: acme-org" + "\n" +
+			"spec:" + "\n" +
+			"  podSelector:" + "\n" +
+			"    matchLabels:" + "\n" +
+			"  ingress:" + "\n" +
+			"  - from:" + "\n" +
+			"    - podSelector: {}"
+		filename := "temp.yaml"
+		WriteToFile(policy, filename)
+
+		MustRun("kubectl apply -f", filename)
+		// ensure this policy can be propagated to its descendants
+		MustRun("kubectl hns config set-type --apiVersion networking.k8s.io/v1 --kind NetworkPolicy propagate")
+		expected := []string {
+			nsOrg 	+ "    deny-from-other-namespaces",
+			nsService1 + "   deny-from-other-namespaces",
+			nsService2 + "   deny-from-other-namespaces",
+			nsTeamA + "      deny-from-other-namespaces",
+			nsTeamB + "      deny-from-other-namespaces",
+		}
+		RunShouldContainMultiple(expected, 2, "kubectl get netpol --all-namespaces")
+
+		// Now we’ll see that we can no longer access service-2 from client a client in service-1:
+		RunShouldContainAndExitWithError("wget: download timed out", 10, 
+			"kubectl run client -n", nsService1, "-i --image=alpine --restart=Never --rm -- sh -c", "\"wget -qO- --timeout 2 http://s2.service-2\"")
+		
+		// create a second network policy that will allow all namespaces within team-a to be able to communicate with each other
+		policy = "" +
+			"kind: NetworkPolicy" + "\n" +
+			"apiVersion: networking.k8s.io/v1" + "\n" +
+			"metadata:" + "\n" +
+			"  name: allow-team-a" + "\n" +
+			"  namespace: team-a" + "\n" +
+			"spec:" + "\n" +
+			"  podSelector:" + "\n" +
+			"    matchLabels:" + "\n" +
+			"  ingress:" + "\n" +
+			"  - from:" + "\n" +
+			"    - namespaceSelector:" + "\n" +
+			"        matchExpressions:" + "\n" +
+			"          - key: 'team-a.tree.hnc.x-k8s.io/depth'" + "\n" +
+			"            operator: Exists"
+		WriteToFile(policy, filename)
+		MustRun("kubectl apply -f" + filename)
+		expected = []string {
+			nsService1 + "   allow-team-a",
+			nsService2 + "   allow-team-a",
+			nsTeamA + "      allow-team-a",
+		}
+		RunShouldContainMultiple(expected, 2, "kubectl get netpol --all-namespaces")
+
+		// Now, we can access the service from other namespaces in team-a, but not outside of it:
+		RunShouldContain("Welcome to nginx!", 10, 
+			"kubectl run client -n", nsService1, "-i --image=alpine --restart=Never --rm -- sh -c", "\"wget -qO- --timeout 2 http://s2.service-2\"")
```

Comment:
This would be more more clear if it looked like this:

`RunShouldContain("Welcome to nginx!", 10, "kubectl run client -n", nsService1, clientArgs)`

Reply:
done